### ISSUE TITLE
`MSBuildExtensionsPath` appends a `\` backslash in .NET Core mode.

### ISSUE THREAD

**Issue by Nirmal4G** [Microsoft/msbuild#3045]
### Steps to reproduce

Set to any simple project in a directory and execute `msbuild` and `dotnet msbuild`

```cmd
:: .NET Framework edition
msbuild /bl /t:Restore
```
```cmd
:: .NET Core edition
dotnet msbuild /bl /t:Restore
```

Examine the `.binlog` file using [`MSBuild Structured Log Viewer`](/KirillOsenkov/MSBuildStructuredLog).

#### .NET Core mode:
![.NET Core mode](https://user-images.githubusercontent.com/4962580/36947255-633c20ba-1fef-11e8-84d8-28beb1bd9d18.png)

#### .NET Framework mode:
![.NET Framework mode](https://user-images.githubusercontent.com/4962580/36947313-24179a26-1ff0-11e8-8d31-22f436384016.png)

### Expected  behavior
`MSBuildExtensionsPath` should not have `\` backslash at the end!

### Actual behavior
`MSBuildExtensionsPath` has `\` backslash at the end only when run in .NET Core mode!

### Environment data
`dotnet msbuild /version`:
```
Microsoft (R) Build Engine version 15.5.180.51428 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

15.5.180.51428
```

`msbuild /version`:
```
Microsoft (R) Build Engine version 15.5.180.51428 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

15.5.180.51428
```

OS info:

`dotnet --info`
```
.NET Command Line Tools (2.1.4)

Product Information:
 Version:            2.1.4
 Commit SHA-1 hash:  5e8add2190

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.16299
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.4\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.5
  Build    : 17373eb129b3b05aa18ece963f8795d65ef8ea54
```

**Comment by Nirmal4G** [comment]
cc @rainersigwald According to the docs it shouldn't happen, so is it a bug on just .NET Core side?

**Comment by Nirmal4G** [comment]
v15.9, it's still there...

**Comment by Nirmal4G** [comment]
v16.1, It's still there...

**Comment by rainersigwald** [comment]
Can you elaborate on what problem you're observing as a result of this?

**Comment by Nirmal4G** [comment]
For my scenarios, I use custom sdks in many of my projects. When authoring some of them I found that, [in the docs](https://docs.microsoft.com/visualstudio/msbuild/msbuild-reserved-and-well-known-properties), `MSBuildExtensionsPath` should not have `\` backslash at the end. So, my and possibly others' targets use it like this: `$(MSBuildExtensionsPath)SomeFolder\SomeBuildProcess.targets`.

When those targets gets shared, let's say, by using `msbuild.exe` instead of `dotnet (ms)build` it breaks the build process, since the path becomes like this: `C:\Program Files\MSBuildSomeFolder\SomeBuildProcess.targets`.

Basically, this a **huge behavioral change**, sure people are not directly using `MSBuildExtensionsPath` anymore, but I've seen code in wild, that leads to many build problems, especially for new people familiarizing with `dotnet msbuild` and authoring custom sdks!

**Comment by rainersigwald** [comment]
I remain confused, thanks for being patient with me.

If the spec is that it does not have a trailing slash, wouldn't you need to always call it with a slash? To use your example `$(MSBuildExtensionsPath)\SomeFolder\SomeBuildProcess.targets`?

**Comment by Nirmal4G** [comment]
It's because `.NET Core` version don't have the slash. And most people are not looking at the docs, they are learning these from the tools they installed, so if they learn from the tool (`.NET Core SDK`) that's displaying wrong value, instead of what's in the spec, they write incompatible code, inturn makes the build break!

**Comment by Nirmal4G** [comment]
I fixed the issue, in my local build. Can I open a PR?
